pkgs/development/python-modules/ifcopenshell/default.nix

```diff
@@ -0,0 +1,67 @@
+{ stdenv
+, buildPythonPackage
+, fetchFromGitHub
+, substituteAll
+, python
+, gcc10
+, cmake
+, boost172
+, icu
+, swig
+, pcre
+, opencascade-occt
+, opencollada
+, libxml2
+}:
+
+buildPythonPackage rec {
+  pname = "ifcopenshell";
+  version = "0.6.0b0";
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner  = "IfcOpenShell";
+    repo   = "IfcOpenShell";
+    rev    = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "1ad1s9az41z2f46rbi1jnr46mgc0q4h5kz1jm9xdlwifqv9y04g1";
+  };
+
+  nativeBuildInputs = [ gcc10 cmake ];
+
+  buildInputs = [
+    boost172
+    icu
+    pcre
+    libxml2
+  ];
+
+  patches = [
+    (substituteAll {
+      name = "site-packages.patch";
+      src = ./site-packages.patch;
+      site_packages = "lib/${python.libPrefix}/site-packages";
+    })
+  ];
+
+  preConfigure = ''
+    cd cmake
+  '';
+
+  cmakeFlags = [
+    "-DOCC_INCLUDE_DIR=${opencascade-occt}/include/opencascade"
+    "-DOCC_LIBRARY_DIR=${opencascade-occt}/lib"
+    "-DOPENCOLLADA_INCLUDE_DIR=${opencollada}/include/opencollada"
+    "-DOPENCOLLADA_LIBRARY_DIR=${opencollada}/lib/opencollada"
+    "-DSWIG_EXECUTABLE=${swig}/bin/swig"
+    "-DLIBXML2_INCLUDE_DIR=${libxml2.dev}/include/libxml2"
+    "-DLIBXML2_LIBRARIES=${libxml2.out}/lib/libxml2.so"
```

Comment:
This line makes it fail on darwin:

```
make[2]: *** No rule to make target '/nix/store/02008j3yn1hs92wncj170j3przajm7j9-libxml2-2.9.10/lib/libxml2.so', needed by 'examples/IfcParseExamples'.  Stop.
make[2]: *** Waiting for unfinished jobs....
```

```
 tree /nix/store/02008j3yn1hs92wncj170j3przajm7j9-libxml2-2.9.10/lib/
/nix/store/02008j3yn1hs92wncj170j3przajm7j9-libxml2-2.9.10/lib/
├── libxml2.2.dylib
├── libxml2.dylib -> libxml2.2.dylib
└── libxml2.la

0 directories, 3 files
```

It should probably be ``libxml2.dylib`` on darwin.

Reply:
I used `if stdenv.isDarwin then "libxml2.dylib" else "libxml2.so"` which I could test under linux and hopefully works with darwin.